shared/domains/chelonia/utils.js

```diff
@@ -278,18 +279,16 @@ export const keyAdditionProcessor = function (hash: string, keys: (GIKey | Encry
         key.meta.private.content &&
         !sbp('chelonia/haveSecretKey', key.id, !key.meta.private.transient)
       ) {
-        try {
-          decryptedKey = key.meta.private.content.valueOf()
-          decryptedKeys.push([key.id, decryptedKey])
-          storeSecretKey(key, decryptedKey)
-        } catch (e) {
-          console.warn(`Secret key decryption error '${e.message || e}':`, e)
-          // Ricardo feels this is an ambiguous situation, however if we rethrow it will
-          // render the contract unusable because it will undo all our changes to the state,
-          // and it's possible that an error here shouldn't necessarily break the entire
-          // contract. For example, in some situations we might read a contract as
-          // read-only and not have the key to write to it.
+        const decryptedKeyResult = unwrapMaybeEncryptedData(key.meta.private.content)
+        // Unable to decrypt
+        if (!decryptedKeyResult) continue
+        // Data aren't encrypted
+        if (decryptedKeyResult.encryptionKeyId == null) {
+          throw new Error('Expected encrypted data but got unencrypted data for key with ID: ' + key.id)
         }
+        decryptedKey = decryptedKeyResult.data
+        decryptedKeys.push([key.id, decryptedKey])
+        storeSecretKey(key, decryptedKey)
```

Comment:
This could be the reason of the following error.

https://okturtles.slack.com/archives/C0EH7P20Y/p1710209299493659